src/App.jsx

```diff
@@ -229,16 +235,41 @@ function KanbanBoard() {
                                   <div
                                     ref={provided.innerRef}
                                     {...provided.draggableProps}
-                                    {...(user ? provided.dragHandleProps : {})}
+                                    {...provided.dragHandleProps}
                                     className={`task ${snapshot.isDragging ? 'dragging' : ''}`}
                                     style={{
                                       ...provided.draggableProps.style,
-                                      cursor: user ? 'grab' : 'default'
                                     }}
-                                    onClick={() => handleTaskClick(column.id, task)}
+onClick={useCallback((e) => {
+  if (!snapshot.isDragging) {
+    handleTaskClick(column.id, task);
+  }
+}, [column.id, task, handleTaskClick, snapshot.isDragging])}
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
const handleClick = useCallback(() => {
  if (!snapshot.isDragging) {
    handleTaskClick(column.id, task);
  }
}, [column.id, task, handleTaskClick, snapshot.isDragging]);

// In JSX:
onClick={handleClick}
```
Consider extracting the memoized callback into a separate variable for better readability and potential reuse. This also makes the JSX more concise.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;